src/audio-manager.ts

```diff
@@ -0,0 +1,208 @@
+ * @example
+ * ```
+ * async start() {
+ *      this.audio = await AudioManager.load('click.wav');
+ * }
+ * onPress() {
+ *      this.audio.play();
+ * }
+ * // if not needed any longer
+ * this.audio.destroy();
+ * ```
+ */
+export const AudioManager = {
+    async load(source: string): Promise<PlayableNode> {
+        if (_bufferCache.has(source)) {
+            const [audioBuffer, referenceCount] = _bufferCache.get(source)!;
+            _bufferCache.set(source, [audioBuffer, referenceCount + 1]);
+            return new PlayableNode(source, audioBuffer);
+        }
+
+        try {
+            const response = await fetch(source);
+            const arrayBuffer = await response.arrayBuffer();
+            const audioBuffer = await _audioContext.decodeAudioData(arrayBuffer);
+
+            _bufferCache.set(source, [audioBuffer, 1]);
+            return new PlayableNode(source, audioBuffer);
+        } catch (error) {
+            throw error;
+        }
+    },
+};
+
+/* AudioContext only unlocks on user interaction, so we wait until the user interacted and the resume */
+async function _unlockAudioContext(): Promise<void> {
+    return new Promise<void>((resolve) => {
+        const unlockHandler = () => {
+            _audioContext.resume().then(() => {
+                window.removeEventListener('click', unlockHandler);
+                window.removeEventListener('touch', unlockHandler);
+                window.removeEventListener('keydown', unlockHandler);
+                window.removeEventListener('mousedown', unlockHandler);
+                resolve();
+            });
+        };
+
+        window.addEventListener('click', unlockHandler);
+        window.addEventListener('touch', unlockHandler);
+        window.addEventListener('keydown', unlockHandler);
+        window.addEventListener('mousedown', unlockHandler);
+    });
+}
+
+/**
+ * Represents a playable audio node that can be used to play audio panned or without panning.
+ *
+ * @note Use the `destroy()` method if audio is not going to be used anymore, to avoid unused audio files
+ * clogging up memory
+ */
+class PlayableNode {
+    /** Whether to loop the audio */
+    public loop: boolean = false;
+
+    /** Whether to enable HRTF over regular panning */
+    public HRTF: boolean = false;
+
+    private _audioBuffer: AudioBuffer;
+    private _source: string;
+    private _isPlaying: boolean = false;
+    private _gainNode: GainNode = new GainNode(_audioContext);
+    private _pannerNode: PannerNode | undefined;
+    private _audioNode: AudioBufferSourceNode = new AudioBufferSourceNode(_audioContext);
+    private _destroy: boolean = false;
+
+    constructor(src: string, audioBuffer: AudioBuffer) {
+        this._audioBuffer = audioBuffer;
+        this._source = src;
+        this._gainNode.connect(_audioContext.destination);
+    }
+
+    /**
+     * Asynchronously plays the loaded audio. If the audio is already playing, it stops the current playback and starts anew.
```

Comment:
Missing space between `a` and `new`

Reply:
anew is a word, and means as much as "from the beginning". I've changed it to that for clearity.

Reply:
TIL. When I am wrong you can just keep it ☺️ Not bad thing to improve my vocabulary